presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -516,7 +520,9 @@ protected Boolean visitFieldReference(FieldReference node, Void context)
                 return true;
             }
 
-            FieldId fieldId = requireNonNull(columnReferences.get(NodeRef.<Expression>of(node)), "No FieldId for FieldReference");
+            List<FieldId> fieldIdList = columnReferences.get(NodeRef.<Expression>of(node));
+            checkState(fieldIdList.size() == 1, "No FieldId for FieldReference");
+            FieldId fieldId = fieldIdList.get(0);
```

Comment:
Add a getter for this instead of duplicating this check multiple places.

Reply:
Done